dawn/test/unit-test/dawn/IIR/TestExtent.cpp

```diff
@@ -19,116 +19,122 @@ using namespace dawn;
   extents.merge(extentsToMerge);
 
-  EXPECT_TRUE((extents[0] == Extent{-2, -1}));
-  EXPECT_TRUE((extents[1] == Extent{1, 2}));
-  EXPECT_TRUE((extents[2] == Extent{0, 0}));
+  compareExtents(extents, {-2, -1, 1, 2, 0, 0});
 }
 
 TEST(ExtentsTest, Merge3) {
-  Extents extents(ast::Offsets{ast::cartesian, -1, 1, 0});
+  Extents extents(dawn::ast::cartesian_{}, ast::Offsets{ast::cartesian, -1, 1, 0});
   extents.merge(ast::Offsets{ast::cartesian, -2, 0, 0});
 
-  EXPECT_TRUE((extents[0] == Extent{-2, -1}));
-  EXPECT_TRUE((extents[1] == Extent{0, 1}));
-  EXPECT_TRUE((extents[2] == Extent{0, 0}));
+  compareExtents(extents, {-2, -1, 0, 1, 0, 0});
 }
 
 TEST(ExtentsTest, Add1) {
-  Extents extents(-1, 1, 0, 0, 0, 0);
-  extents.add(ast::Offsets{ast::cartesian, 1, 0, 0});
+  // TODO check if test is only place where add(Offset& ) is used
+  // Extents extents(dawn::ast::cartesian_{}, -1, 1, 0, 0, 0, 0);
+  // extents.add(ast::Offsets{ast::cartesian, 1, 0, 0});
 
-  EXPECT_TRUE((extents[0] == Extent{0, 2}));
-  EXPECT_TRUE((extents[1] == Extent{0, 0}));
-  EXPECT_TRUE((extents[2] == Extent{0, 0}));
+  // EXPECT_TRUE((extents[0] == Extent{0, 2}));
+  // EXPECT_TRUE((extents[1] == Extent{0, 0}));
+  // EXPECT_TRUE((extents[2] == Extent{0, 0}));
 }
 
 TEST(ExtentsTest, Add2) {
-  Extents extents(0, 1, 0, 0, 0, 0);
-  extents.add(ast::Offsets{ast::cartesian, -1, 0, 0});
+  // Extents extents(0, 1, 0, 0, 0, 0);
+  // extents.add(ast::Offsets{ast::cartesian, -1, 0, 0});
 
-  EXPECT_TRUE((extents[0] == Extent{-1, 0}));
-  EXPECT_TRUE((extents[1] == Extent{0, 0}));
-  EXPECT_TRUE((extents[2] == Extent{0, 0}));
+  // EXPECT_TRUE((extents[0] == Extent{-1, 0}));
+  // EXPECT_TRUE((extents[1] == Extent{0, 0}));
+  // EXPECT_TRUE((extents[2] == Extent{0, 0}));
 }
 
 TEST(ExtentsTest, Add3) {
-  Extents extents(-2, 2, 0, 0, 0, 0);
-  extents.add(ast::Offsets{ast::cartesian, 1, 0, 0});
+  // Extents extents(-2, 2, 0, 0, 0, 0);
+  // extents.add(ast::Offsets{ast::cartesian, 1, 0, 0});
 
-  EXPECT_TRUE((extents[0] == Extent{-1, 3}));
-  EXPECT_TRUE((extents[1] == Extent{0, 0}));
-  EXPECT_TRUE((extents[2] == Extent{0, 0}));
+  // EXPECT_TRUE((extents[0] == Extent{-1, 3}));
+  // EXPECT_TRUE((extents[1] == Extent{0, 0}));
+  // EXPECT_TRUE((extents[2] == Extent{0, 0}));
 }
 
 TEST(ExtentsTest, Add4) {
-  Extents extents(-2, 2, 0, 0, 0, 0);
-  extents.add({-2, 2, 0, 0, 0, 0});
+  Extents extents(dawn::ast::cartesian_{}, -2, 2, 0, 0, 0, 0);
+  auto addedExtents =
+      dawn::iir::Extents::add(extents, Extents(dawn::ast::cartesian_{}, -2, 2, 0, 0, 0, 0));
 
-  EXPECT_TRUE((extents[0] == Extent{-4, 4}));
-  EXPECT_TRUE((extents[1] == Extent{0, 0}));
-  EXPECT_TRUE((extents[2] == Extent{0, 0}));
+  compareExtents(addedExtents, {-4, 4, 0, 0, 0, 0});
 }
 
 TEST(ExtentsTest, Add5) {
-  Extents extents({-2, 2, 0, 4, 0, 0});
-  extents.add(ast::Offsets{ast::cartesian, 2, 2, 3});
+  // Extents extents({-2, 2, 0, 4, 0, 0});
```

Comment:
what about all these commented out tests?

Reply:
adding an offset is _only_ used by these tests but nowere else

Reply:
will remove it completely